subsquid/src/server-extension/resolvers/pabloDaily.ts

```diff
@@ -77,13 +81,18 @@ export class PabloDailyResolver implements ResolverInterface<PabloDaily> {
       return acc;
     }, {});
 
-    const totalVolumes = Object.keys(swapsMap).map(
-      assetId =>
+    const totalVolumes: Array<PoolAmount> = [];
+
+    for (const assetId of Object.keys(swapsMap)) {
+      const price = await getOrCreateAssetPrice(manager, assetId, new Date().getTime());
```

Comment:
I think this should be processed s.t. 
1. each price is checked first
2. Get an array of timestamps that are missing
3. request all of the missing ones at once with [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) and apply the processing mechanism to all of them as they arrive

This will probably be much more efficient code

Reply:
however since we're on the clock let's get this merged and apply this going forward

Reply:
ah makse sense - looks like I was also referencing the wrong code block lol.
Gotcha sir 🫡